### PR TITLE
Enable REST API by default in template configuration

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Enable REST API by default in template configuration, so that e.g. `hz-healthcheck` script can use it.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [x] Send backports/forwardports if fix needs to be applied to past/future releases
- [x] New public APIs have `@Nonnull/@Nullable` annotations
- [x] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

File: hazelcast/src/main/config-template/hazelcast-assembly.xml [hunk 1]
```diff
@@ -174,7 +174,7 @@
         <!--
             Configuration for REST endpoints. When it's enabled, it will allow healthcheck calls over the HTTP protocol.
```
Comment (kwart) [inline hazelcast/src/main/config-template/hazelcast-assembly.xml:175]:
```suggestion
            Configuration for REST endpoints. It allows healthcheck calls over the HTTP protocol.
```

File: hazelcast/src/main/config-template/hazelcast-assembly.yaml [hunk 2]
```diff
@@ -113,7 +113,7 @@ hazelcast:
         enabled: false
     # Configuration for REST endpoints. When it's enabled, it will allow healthcheck calls over the HTTP protocol.
```
Comment (kwart) [inline hazelcast/src/main/config-template/hazelcast-assembly.yaml:114]:
```suggestion
    # Configuration for REST endpoints. It allows healthcheck calls over the HTTP protocol.
```

**Review by kwart — APPROVED** [review]
Approving and suggesting also comments update.

**Review by ldziedziul — COMMENTED** [review]
I think we should update this as well: https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731

**Comment by TomaszGaweda** [comment]
> I think we should update this as well:
> 
> https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731

True, updated as well, also xml config for 5.2

**Comment by kwart** [comment]
> > I think we should update this as well:
> > https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731
> 
> True, updated as well, also xml config for 5.2

This is not correct, the default value is `false`. We explicitly set it to `true` in some distributions. If the value is not specified, the REST has to remain disabled.

**Review by kwart — CHANGES_REQUESTED** [review]
> > I think we should update this as well:
> > https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731
> 
> True, updated as well, also xml config for 5.2

This is not correct, the default value is `false`. We explicitly set it to `true` in some distributions. If the value is not specified, the REST has to remain disabled.

**Comment by TomaszGaweda** [comment]
> > > I think we should update this as well:
> > > https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731
> > 
> > 
> > True, updated as well, also xml config for 5.2
> 
> This is not correct, the default value is `false`. We explicitly set it to `true` in some distributions. If the value is not specified, the REST has to remain disabled.

Thanks for clarification, updated :)

**Comment by TomaszGaweda** [comment]
run-lab-run

**Review by kwart — APPROVED** [review]

**Comment by TomaszGaweda** [comment]
run-lab-run

**Pull request merged**